tensorflow/python/kernel_tests/reduction_ops_test.py

```diff
@@ -435,7 +435,7 @@ def testFloatReduce3D(self):
   def testGradient(self):
     s = [2, 3, 4, 2]
     # NOTE(kearnes): divide by 20 so product is a reasonable size
-    x = np.arange(1.0, 49.0).reshape(s).astype(np.float32) / 20.
+    x = np.arange(0.0, 48.0).reshape(s).astype(np.float32) / 20.
```

Comment:
Good idea, I'll implement that together with an array that doesn't have a zero, as arrays with a zero have a very simple gradient which could prevent bugs from showing up.